lib/Value/WeBWorK.pm

```diff
@@ -73,6 +73,11 @@ sub Value::Formula::PGseedRandom {
 }
 sub Value::Formula::PGgetRandom {shift->{PGrandom}->random(@_)}
 
+sub Value::maketext {
+	my $envir =eval('$main::rh_envir');
```

Comment:
I think so.  The Value::WeBWorK.pm file is evaluated at compile time when the child is launched.  Each request is handled within a new Safe compartment which changes the value of $main (in addition the contents of the environment variable $rh_envir is different for every request). 

This is also how Davide handled the initialization of $envir for context  a few lines above in WeBWorK.pm
---------------------------------------
sub Parser::Context::copy {
  my $self = shift;
  my $context = Value::Context::copy($self,@_);
  return $context unless $Parser::installed;  # only do WW initialization after parser is fully loaded
  return $context if $context->{WW} && scalar(keys %{$context->{WW}}) > 0;
  my $envir = eval('\\%main::envir');
  return $context unless $envir && scalar(keys(%{$envir})) > 0;
  my $ww = $context->{WW} = {}; push @{$context->{data}{values}}, 'WW';
  return $context if $Value::_no_WeBWorK_; # hack for command-line debugging
  foreach my $x (@wwEvalFields) {$context->{WW}{$x} = $envir->{$x}}
  $context->flags->set(
     tolerance      => $ww->{numRelPercentTolDefault} / 100,
     zeroLevel      => $ww->{numZeroLevelDefault},
     zeroLevelTol   => $ww->{numZeroLevelTolDefault},
     num_points     => $ww->{functNumOfPoints} + 2,
     max_adapt      => $ww->{functMaxConstantOfIntegration},
     useBaseTenLog  => $ww->{useBaseTenLog},
  );
  $context->{format}{number} = $ww->{numFormatDefault} if $ww->{numFormatDefault} ne '';
  $context;
}